### DISCUSSION TITLE
Transformation: Join by field (inner join)

### DISCUSSION THREAD

**Discussion by patlachance** [grafana/grafana#38150] [Dashboards, Panels & Visualizations]
**What would you like to be added**:

**Inner** join in transformation by one (or many as in #25793) field(s).

**Why is this needed**:

[Inner join](https://www.diffen.com/difference/Inner_Join_vs_Outer_Join) allow selecting only data that are present in both/all datasources (intersection).
Outer join already available seems to be implemented as a full outer join, which returns all data from both/all datasources(union).

Our initial use case would be exporting query filters in external database. For example, select only rows from Prometheus datasource that have a matching line a second datasource (ex: Postgres, MySQL, ...).

***Example***:

*DB Table namespace_mapping*
|id|user|namespace|
|-|-|-|
| 1 | john | namespace1 |
| 2 | doe | namespace2 |
| 3 | john | namespace3 |

*DB Query*: SELECT namespace FROM namespace_mapping WHERE user = 'john';

*Prometheus dataset*
|id|namespace|metric1|metric2
|-|-|-|-|
| 1 | namespace1 | 22 | 45 |
| 2 | namespace2 | 13 | 56 |
| 3 | namespace3 | 31 | 19 |

*Expected result*

|namespace|metric1|metric2
|-|-|-|
| namespace1 | 22 | 45 |
| namespace3 | 31 | 19 |

**Comment by patlachance** [comment]
@marefr it it not a duplicate of #25793, which request the possibility to join on multiple fields using **outer join**. This issue is for **inner join** and it is not included in #25793. Please reopen

**Comment by marefr** [comment]
Thanks. I reopened it.

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by tlifschitz** [comment]
+1 to this feature, it would be really useful when having multiple data sources

**Comment by allfro** [comment]
Yes. This is extremely valuable when we want to just use a lookup table that can be used as a value mapping for a specific column!

**Comment by jonahmons** [comment]
In the meantime, is there any way to work around this? 

+1 And yes, it would be nice to have this feature.

**Comment by liubofeng** [comment]
In the meantime, is there any way to work around this?

+1 And yes, it would be nice to have this feature.

**Reply by iceman91176** [reply]
In the meantime, is there any way to work around this?

+1 And yes, it would be nice to have this feature.

**Comment by leeoniya** [comment]
this is now available. see these PRs:

https://github.com/grafana/grafana/pull/53865
https://github.com/grafana/grafana/pull/53913
